plugin/core/views.py

```diff
@@ -172,3 +176,39 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
         "options": formatting_options(view.settings()),
         "range": region_to_range(view, region).to_lsp()
     })
+
+
+def minihtml(content: Union[str, dict], view: sublime.View) -> str:
+    """ Content can be a string or MarkupContent """
+    if isinstance(content, str):
+        return text2html(content)
+    elif isinstance(content, dict):
+        value = content.get("value") or ""
+        kind = content.get("kind")
+        if kind == "markdown":
+            return mdpopups.md2html(view, value)
+        else:
+            # must be plaintext
+            return text2html(value)
+    else:
+        return ''
+
+
+def text2html(content: str) -> str:
+    content = "\n".join(textwrap.wrap(content, width=54,
+                                      replace_whitespace=False, break_long_words=False, break_on_hyphens=False))
+    content = html.escape(content).replace('\n', '<br>').replace('\t', '&nbsp;&nbsp;&nbsp;&nbsp;')
+
+    def replace_npbs(match: Any) -> str:
```

Comment:
Doesn't md2html already have this logic builtin? What am I missing?

Also should the method name be `replace_nbsp`?

Reply:
Same as below I presume

Reply:
Darn it :) https://github.com/sublimelsp/LSP/pull/1022/commits/6058b346788c2108466908592182952421cb9c32 
Thanks